src/Symfony/Component/Config/Definition/EnumNode.php

```diff
@@ -78,13 +104,48 @@ protected function finalizeValue(mixed $value): mixed
     {
         $value = parent::finalizeValue($value);
 
-        if (!\in_array($value, $this->values, true)) {
-            $ex = new InvalidConfigurationException(\sprintf('The value %s is not allowed for path "%s". Permissible values: %s', json_encode($value), $this->getPath(), $this->getPermissibleValues(', ')));
-            $ex->setPath($this->getPath());
+        if ($this->enumFqcn) {
```

Comment:
maybe we can turn this into an early return condition to reduce nesting?
```suggestion
        if (!$this->enumFqcn) {
```